### PR TITLE
pkg/trace/api: evp proxy support for app keys

### PR DESCRIPTION
**Pull request by albertvaka**
### What does this PR do?

Allows the EVPProxy to be used against Datadog APIs that require an Application Key in addition to an API Key.

### Motivation

Needed for an upcoming CI Visibility product.

### Additional Notes

* This is not the first usage of an app key in the Agent, the security agent is [already using them](https://github.com/DataDog/datadog-agent/blob/b3df4adca9362d677bcbdce59a2e641fe4bcac8c/cmd/security-agent/app/runtime.go#L692).
* I haven't added support for dual-shipping using two app keys. It would require a larger change and we don't need it as of now.

### Describe how to test/QA your changes

Can be tested the same way as the PR where the EVP proxy was introduced: https://github.com/DataDog/datadog-agent/pull/12271

### CODE REVIEW

File: pkg/trace/config/config.go [hunk 1]
```diff
@@ -266,6 +266,8 @@ type EVPProxy struct {
 	DDURL string
 	// APIKey is the main API Key (defaults to the main API key).
 	APIKey string
+	// APIKey is the main API Key (defaults to the main API key).
```
Comment (ajgajg1134) [inline pkg/trace/config/config.go:269]: Should this comment be changed? At least it should say `ApplicationKey`

File: pkg/trace/api/evp_proxy.go [hunk 2]
```diff
@@ -118,6 +118,7 @@ func (t *evpProxyTransport) RoundTrip(req *http.Request) (rresp *http.Response,
 	containerID := req.Header.Get(headerContainerID)
 	contentType := req.Header.Get("Content-Type")
 	userAgent := req.Header.Get("User-Agent")
+	needsAppKey := (req.Header.Get("X-Datadog-NeedsAppKey") == "true")
```
Comment (ajgajg1134) [inline pkg/trace/api/evp_proxy.go:121]: any concerns about case-sensitivity here?

**Review by ajgajg1134 — CHANGES_REQUESTED** [review]

File: pkg/trace/config/config.go (hunk 1 above)
Comment (albertvaka) [reply pkg/trace/config/config.go:269]:
```suggestion
	// ApplicationKey to be used for requests with the X-Datadog-NeedsAppKey set (defaults to the top-level Application Key).
```

File: pkg/trace/api/evp_proxy.go (hunk 2 above)
Comment (albertvaka) [reply pkg/trace/api/evp_proxy.go:121]:
```suggestion
	needsAppKey := (req.Header.Get("X-Datadog-NeedsAppKey").ToLower() == "true")
```

**Review by ajgajg1134 — APPROVED** [review]
Looks good!

**Pull request merged**